projects/packages/wp-js-data-sync/src/endpoints/class-action-endpoint.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Register and handle custom action REST API Endpoints for data sync entries.
+ *
+ * @package automattic/jetpack-wp-js-data-sync
+ */
+
+namespace Automattic\Jetpack\WP_JS_Data_Sync\Endpoints;
+
+use Automattic\Jetpack\WP_JS_Data_Sync\Contracts\Data_Sync_Action;
+
+class Action_Endpoint {
+
+	/**
+	 * @var Data_Sync_Action  - The class that handles the action.
+	 */
+	private $action_class;
+	/**
+	 * @var string $rest_namespace - The namespace for the REST API endpoint.
+	 */
+	private $rest_namespace;
+
+	/**
+	 * @var string $route - The route for the REST API endpoint.
+	 */
+	private $route;
+
+	/**
+	 * @var Authenticated_Nonce $nonce - The nonce for the REST API endpoint.
+	 */
+	private $nonce;
+
+	/**
+	 * @param $namespace
+	 * @param $route
+	 * @param $action_class
+	 */
+	public function __construct( $namespace, $key, $action_name, $action_class ) {
+		$this->action_class   = $action_class;
+		$this->rest_namespace = $namespace;
+		$this->route          = $key . '/action/' . $action_name;
+		$this->nonce          = new Authenticated_Nonce( "{$namespace}_{$this->route}_action" );
```

Comment:
I just noticed, that this class is also present in the `Async_Option` package.

Reply:
Oh, that's just the inspect plugin that's the origin of DataSync. It needs to get updated.